src/validate/editor.js

```diff
@@ -0,0 +1,20 @@
+'use babel'
+
+import throwIfFail from './throw'
+
+export const isValidPoint = (textBuffer, line, column) => {
+  // Clip the given point to a valid one
+  const validPoint = textBuffer.clipPosition([line, column])
+  // Compare to original
+  return validPoint.isEqual([line, column])
+}
+
+export const throwIfInvalidPoint = (tb, line, col) => throwIfFail(
+  (`${line}:${col} isn't a valid point!`),
+  isValidPoint(tb, line, col)
+)
+
+export const isValidScope = (editor, scopes) => editor.getCursors()
```

Comment:
I think I'd say `hasValidScope` is a better name here, the function is checking if the `TextEditor` _has_ a cursor with some scope that matches the list of valid scopes passed in.

Reply:
Let's rename `scopes` to `validScopes` to make it clear what's going on.

Reply:
Makes sense. Actually I think that little changes makes the entire function noticeably easier to make sense of. 👍 👍